File: Python/AOC2015/days/day1.py
class Day1:
    def __init__(self):
        with open("../../Data/2015/day1.txt") as f:
            self.data = f.read()
    
    def find_floor(self):
        floor = 0
        index = 1
        not_printed = True
        for x in self.data:
            if x == '(':
                floor += 1
            else:
                floor -= 1
            if floor == -1 and not_printed:
                not_printed = False
                print(f"The basement is entered at index {index}.")
            index += 1
        print(f"The final floor number was {floor}.")

File: CSharp/AOC2015/Days/Day1.cs
namespace AOC2015.Days;

public class Day1
{
    private readonly string _data = File.ReadAllText("../../Data/2015/day1.txt");
    private bool _basementEntered = false;
    private int _floor = 0;
    private int _timeBasementEntered = 1;

    public void RunLift()
    {
        foreach (char c in _data)
        {
            _floor += c == '(' ? 1 : -1;

            if (_floor == -1)
                _basementEntered = true;
            if (!_basementEntered)
                _timeBasementEntered++;
        }

        Console.WriteLine($"The final floor was {_floor}, and the basement was first entered at {_timeBasementEntered}.");
    }
}